File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    one,two,three = 0,0,0
    for i in range(len(answers)):
        if (i+1)%5 == 0:
            ans = 5
        else:
            ans = (i+1)%5
        if ans == answers[i]:
            one+=1
    for i in range(len(answers)):
        if i%2==0:
            ans = 2
        elif i%8==1:
            ans = 1
        elif i%8==3:
            ans = 3
        elif i%8==5:
            ans = 4
        elif i%8==7:
            ans = 5
        if ans == answers[i]:
            two+=1
    for i in range(len(answers)):
        if i%10<2:
            ans = 3
        elif i%10<4:
            ans = 1
        elif i%10<6:
            ans = 2
        elif i%10<8:
            ans = 4
        elif i%10<10:
            ans = 5
        if ans == answers[i]:
            three+=1
    a = max(one,two,three)
    ans = []
    if a == one:
        ans.append(1)
    if a == two:
        ans.append(2)
    if a == three:
        ans.append(3)
    return ans

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    let one=0
    let two=0
    let three=0
    for(let i=0;i<answers.length;i++) {
        if(i%5+1===answers[i]) one++
        if(i%2===0 && answers[i]===2) two++
        else if(i%8===1 && answers[i]===1) two++
        else if(i%8===3 && answers[i]===3) two++
        else if(i%8===5 && answers[i]===4) two++
        else if(i%8===7 && answers[i]===5) two++
        if((i%10===0 || i%10===1) && answers[i]===3) three++
        else if((i%10===2 || i%10===3) && answers[i]===1) three++
        else if((i%10===4 || i%10===5) && answers[i]===2) three++
        else if((i%10===6 || i%10===7) && answers[i]===4) three++
        else if((i%10===8 || i%10===9) && answers[i]===5) three++
    }
    let max = Math.max(one,two,three)
    let ans = []
    let arr = [one,two,three]
    for(let i=0;i<arr.length;i++){
        if(arr[i]===max) ans.push(i+1)
    }
    return ans
}